pkgs/development/libraries/alice-vision/3.0.0.nix

```diff
@@ -0,0 +1,225 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, pkg-config
+
+, pkgs
+
+, assimp
+# use the same boost as in cctag here
+, boost179
+, ceres-solver
+, clp
+, eigen
+, expat
+, flann
+, geogram
+, lemon-graph
+, lz4
+, openexr
+, openimageio
+, zlib
+
+, enableOpenMP ? true
+# Separate openmp input only required on Darwin stdenv (LLVM)
+, openmp
+
+, enableAlembic ? true, alembic
+, enableCctag ? true, cctag
+, enableOpenCV ? true, enableOpenCVContrib ? enableOpenCV, opencv
+}:
+
+let
+  nanoflann = pkgs.nanoflann.overrideAttrs (
+    oldAttrs: rec {
+      version = "1.4.2";
+      src = fetchFromGitHub {
+        owner = "jlblancoc";
+        repo = "nanoflann";
+        rev = "v${version}";
+        hash = "sha256-znIX1S0mfOqLYPIcyVziUM1asBjENPEAdafLud1CfFI=";
+      };
+    }
+  );
+in
+stdenv.mkDerivation rec {
+  pname = "alice-vision";
+  version = "3.0.0";
+
+  outputs = [ "out" "dev" ];
+
+  src = fetchFromGitHub {
+    owner = "alicevision";
+    repo = "AliceVision";
+    rev = "v${version}";
+    hash = "sha256-rFd2AFtvC1RugKGv4tI2k3rtDqXHjdDC3pgOrqDqIT0=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
+  buildInputs = [
+    flann
+    nanoflann
+    openexr
+
+    # Temporary fix until flann 1.9.2 is in Nixpkgs
+    lz4
+  ];
+
+  propagatedBuildInputs = [
+    assimp
+    boost179
+    ceres-solver
+    clp
+    eigen
+    expat
+    geogram
+    lemon-graph
+    openimageio
+    zlib
+  ] ++ lib.optional enableAlembic alembic
+    ++ lib.optional enableCctag cctag
+    ++ lib.optional enableOpenCV opencv
+    ++ lib.optional stdenv.cc.isClang openmp;
+
+  patches = [
+    # Don't optimize for the host processor (could lead to reproducibility issues)
+    ./cmake-disable-optimize-for-architecture.patch
+
+    # Upstream uses patched Clp/Osi/CoinUtils containing CMake build scripts.
+    # Instead of the patched versions, we re-use our packages and add CMake
+    # find modules that use the pkg-config files already generated by those
+    # dependencies.
+    ./find-coin-modules.patch
```

Comment:
I don't know the upstream policy, but in my opinion this patch should be upstreamed, in a "optional mode".

Reply:
Absolutely agree, but I failed to upstream a patch doing something similar a while ago. Upstream actually prefers to maintain their own version of the Coin libraries (see https://github.com/alicevision/AliceVision/pull/1500 and https://github.com/alicevision/AliceVision/issues/1499#issuecomment-1657103811)

Reply:
So, why not use their patched version?

Reply:
Looking further into this, another reasonable way of solving this would be the `coin-or` upstream adopting CMake support, considering that `alice-vision` is not their only user that needs it. This has already been proposed (https://github.com/coin-or/CoinUtils/pull/101, https://github.com/coin-or/Clp/pull/105 and https://github.com/coin-or/Osi/pull/122), but as far as I understand could not yet be merged due to limited developer capacity. Maybe this is how that issue will be resolved some day. Until then, another option for packagers like us might be to look for the best CMake support packages for the Coin libraries in existence, and apply them to all builds, so we would ideally have one build that fits both CMake- and autotools/pkg-config-based reverse dependencies.

Reply:
It looks like future work with upstream. Let's put this in the backlog.